homework-assignments-1/task-2.ts

```diff
@@ -0,0 +1,21 @@
+class User {
+  private _username: string;
+  private _password: string;
+
+  constructor(username: string, password: string) {
+    this._username = username;
+    this._password = password;
+  }
+
+  validateLogin(inputPassword: string) {

```

Comment:
The login validation process should be handled internally by the method without exposing the details of how passwords are compared.

`  public validateLogin(inputPassword: string): boolean {
    // Abstracts the logic of password comparison
    return this.checkPassword(inputPassword);
  }

  // Private method for password comparison
  private checkPassword(inputPassword: string): boolean {
    // Compare the input password with the actual password
    return inputPassword === this.password;
  }`